src/doc/feature/events.dox

```diff
@@ -0,0 +1,130 @@
+// vim:ft=doxygen
+namespace crepe {
+/**
+
+\defgroup feature_events Events
+\ingroup feature
+\brief events
+
+# EventManager
+The `EventManager` is a centralized system for managing event subscriptions and triggering. It facilitates
+decoupled communication between different parts of your application by allowing subscribers to register callbacks
+for specific event types and receive notifications when those events occur.
+## Key Features
+- **Event Subscriptions**: Subscribe to specific events using a unique subscription ID.
+- **Event Channels**: Use channels to scope events, reducing unnecessary callback invocations.
+- **Event Dispatching**: Trigger events immediately or queue them for deferred processing.
+- **Custom Event creation**: Ability to create custom events for easy interaction between scripts
+
+\see EventManager
+\see Events
+
+## Event Types
+The `EventManager` works with a variety of event types derived from the `Event` base class. These include:
+- `KeyPressEvent`: Triggered when a key is pressed.
+- `KeyReleaseEvent`: Triggered when a key is released.
+- `MousePressEvent`: Triggered when a mouse button is pressed.
+- `MouseReleaseEvent`: Triggered when a mouse button is released.
+- `MouseClickEvent`: Triggered when a mouse button is clicked.
+- `MouseMoveEvent`: Triggered when the mouse is moved.
+- `CollisionEvent`: Triggered during object collisions.
+- `TextSubmitEvent`: Triggered when text is submitted.
+- `ShutDownEvent`: Triggered when the application is shutting down.
+
+For detailed documentation on each event type, see the @ref EventTypesFeature "Event Types Feature Page".
+## Usage
+
+### Subscribing to Events
+Use the `subscribe` method to register a callback for a specific event type. The callback will be invoked when
+the event is triggered.
+
+```cpp
+// Example: Subscribing to a KeyPressEvent
+EventManager::get_instance().subscribe<KeyPressEvent>(
+	[](const KeyPressEvent &event) {
+		std::cout << "Key pressed: " << event.key << std::endl;
```

Comment:
Dit compileert niet omdat de handler een boolean moet returnen. Dat wordt ook nergens in dit document uitgelegd.